### PR TITLE
Adding digitalocean-sshkey-id option so that we don't have to create …

### PR DESCRIPTION
**Pull request by usmanismail**
…a new ssh key every time we use machine.

Fixes Issue #1445

### CODE REVIEW

File: drivers/digitalocean/digitalocean.go [hunk 1]
```diff
@@ -172,15 +180,22 @@ func (d *Driver) PreCreateCheck() error {
...
+	if d.SSHKeyID == 0 {
+		log.Infof("Creating SSH key...")
+		key, err := d.createSSHKey()
+		if err != nil {
+			return err
+		}
+		d.SSHKeyID = key.ID
+	} else {
+		log.Infof("Using Key ID: %d", d.SSHKeyID)
+		_, err := d.validateSSHKey(d.SSHKeyID)
```
Comment (ehazlett) [inline drivers/digitalocean/digitalocean.go]: Minor nit: this could be:
```
if _, err := d.validateSSHKey(d.SSHKeyID); err != nil {
    return err
}
```

**Comment by ehazlett** [comment]
Minor nit otherwise looks good.

File: drivers/digitalocean/digitalocean.go [hunk 2]
```diff
@@ -227,6 +241,18 @@ func (d *Driver) Create() error {
 
 	return nil
 }
+func (d *Driver) validateSSHKey(keyId int) (*godo.Key, error) {
+	key, resp, err := d.getClient().Keys.GetByID(keyId)
+	if err != nil {
+		if resp.StatusCode == 404 {
+			return nil, errors.New(fmt.Sprintf("Unable to find ssh key with ID %d", keyId))
```
Comment (nathanleclaire) [inline drivers/digitalocean/digitalocean.go]: Why not use `fmt.Errorf` and eliminate the extra call?

**Comment by usmanismail** [comment]
Any update on this?

**Comment by nathanleclaire** [comment]
Code looks OK.  It needs docs and at least one test (integration or unit).

**Comment by nathanleclaire** [comment]
@usmanismail Any followup on this one?  At least to update the documentation with the new parameters?

**Comment by usmanismail** [comment]
@nathanleclaire Docs updated. There are no tests in there that I can use as a sample/template Other than just basic asserts that CLI parameters were parsed what kind of tests are you looking for.

**Comment by dmp42** [comment]
@usmanismail this unfortunately needs a rebase

**Comment by dgageot** [comment]
@usmanismail It looks like your rebase is more a merge. Can you rebase on top of master please?

**Comment by dgageot** [comment]
Hi @usmanismail any chance you could rebase this PR. Otherwise we might have to close it.

**Comment by usmanismail** [comment]
@dgageot Should be good now.

**Comment by bhurlow** [comment]
@usmanismail @dgageot any update on this? does this still need rebase?

**Comment by dgageot** [comment]
@usmanismail @bhurlow  Sorry, it looks like the code still needs to be rebased. Also, the tests are currently failing.

**Comment by klaemo** [comment]
This would help with #2642 :)

**Comment by dgageot** [comment]
@usmanismail Can you rebase you work please? We'd then be able to review.

**Comment by usmanismail** [comment]
I won't have time to work on this for the next several weeks. Closing for now.

**Comment by bamarni** [comment]
Also looking forward to this :+1:  I just submitted #3144, with a slightly different proposal.

**Comment by ericwooley** [comment]
This would be a really nice feature, installing them manually is a pain

**Pull request closed without merging**